flexget/plugins/clients/flood.py

```diff
@@ -0,0 +1,442 @@
+
+    def delete_torrents(self, config: dict, hashes: list, deleteData: bool = False) -> bool:
+        """
+        Delete a list of torrents in Flood.
+        Will only attempt to delete torrents if the JWT token is not expired or unset.
+        """
+
+        if not hashes:
+            raise PluginError("Parameter 'hashes' cannot be empty.")
+        if self.is_jwt_expired():
+            raise PluginError("Not authenticated with Flood.")
+
+        try:
+            response = post(
+                f"{config['url']}/api/torrents/delete",
+                json={"hashes": hashes, "deleteData": deleteData},
+                cookies={"jwt": self.jwt},
+            )
+
+            if response.status_code == 200:
+                logger.debug(f"Successfully deleted {len(hashes)} torrents from Flood.")
+                return True
+
+            logger.debug(
+                f"Failed to delete {len(hashes)} torrents from Flood. (status={response.status_code})"
+            )
+            return False
+        except RequestException as e:
+            raise PluginError(
+                f"An error occurred while attempting to delete torrents from Flood: {e}"
+            )
+
+
+class InputFlood:
+    """Creates entries from torrents in Flood based
+
+    Example:
+        from_flood:
+            url: http://localhost:3000
+            username: username
+            password: password
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'url': {'type': 'string'},
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+        },
+        'additionalProperties': False,
+        'required': ['url'],
+    }
+
+    def __init__(self) -> None:
+        self.flood = FloodClient()
+
+    def generate_entry(self, config: dict, torrent: dict) -> Entry:
+        """
+        Generate a task entry from a torrent in Flood.
+        """
+
+        entry = Entry(url='', title=torrent['name'])
+
+        for key, value in torrent.items():
+            entry['flood_' + key.lower()] = value
+
+        return entry
+
+    def on_task_input(self, task: Task, config: dict):
+        if self.flood.is_jwt_expired():
+            logger.debug("JWT token is expired. Re-authenticating with Flood.")
+            self.flood.authenticate(config)
+
+        entries = []
+        torrents = self.flood.list_torrents(config)
+
+        for hash, torrent in torrents.items():
+            entry = self.generate_entry(config, torrent)
+
+            if entry:
+                entries.append(entry)
+
+        return entries
+
+
+class OutputFlood:
+    """Add, remove, delete, start, or stop torrents in Flood.
+
+    Example:
+        flood:
+            url: http://localhost:3000
+            username: username
+            password: password
+            path: /downloads/Flexget
+            tags: Flexget
+            action: add
```

Comment:
Again listing the config values and what they represent would be nice.

Reply:
Added descriptions to module docstring examples in Winter/Flexget@c067577